website/server/models/group.js

```diff
@@ -392,7 +392,18 @@ schema.methods.sendChat = function sendChat (message, user) {
   let newMessage = chatDefaults(message, user);
 
   this.chat.unshift(newMessage);
-  this.chat.splice(200);
+
+  const MAX_CHAT_COUNT = 200;
+  const MAX_SUBBED_GROUP_CHAT_COUNT = 400;
+
+  let maxCount = MAX_CHAT_COUNT;
+
+  let plan = this.purchased.plan;
+  if (plan && plan.customerId && !plan.dateTerminated) {
```

Comment:
What I'm saying is that a plan.dateTerminated > now doesn't mean the user has no more subscription benefit, it's the opposite: dateTerminated < now means they've expire > now they still haven't expired and !dateTerminated they still have an active sub (with customerId) or never had one (without customerId)